mixer/cmd/build.go

```diff
@@ -199,113 +218,158 @@ var buildFormatNewCmd = &cobra.Command{
 		if err != nil {
 			fail(err)
 		}
+		// Update mixer to build version +20, to build the bundles with +20 inside not +10
 		if err = b.UpdateMixVer(ver + 20); err != nil {
 			failf("Couldn't update Mix Version")
 		}
 
-		// Set format to format+1 so that the format file inserted into the
-		// update content is the new one
-		newFormat, err := strconv.Atoi(b.State.Mix.Format)
-		if err != nil {
-			fail(err)
-		}
-		newFormat++
-		if err = b.UpdateFormatVersion(strconv.Itoa(newFormat)); err != nil {
-			fail(err)
-		}
-		setWorkers(b)
+		// Change the format internally just for building bundles so the right
+		// /usr/share/defaults/swupd/format is inserted; preserve old format
+		oldFormat := b.State.Mix.Format
+		b.State.Mix.Format = buildFlags.newFormat
 
-		fmt.Println(" Backing up full groups.ini")
-		// Back up groups.ini in case we have deprecated bundles to delete
-		if err = b.CopyFullGroupsINI(); err != nil {
+		// Build bundles normally. At this point the bundles to be deleted should still
+		// be part of the mixbundles list and the groups.ini
+		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
 			fail(err)
 		}
-		// Fill this in w/Update bundle definitions
-		// if err := UpdateBudlesForFormatBump(); err != nil {...}
 
-		// Build the +20 (first build in new format) bundles
-		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
+		// Wipes the <bundle>-info files and replaces them with
+		// <bundle>/ directories that are empty. When the manifest creation happens it will
+		// mark all files in that bundles as deleted.
+		f := func(filename string) error {
```

Comment:
I think declaring this inline makes this long function even longer than it needs to be. Any reason you don't just define it at package scope?

Reply:
'cause por que no

Reply:
because
> declaring this inline makes this long function even longer than it needs to be